### ISSUE TITLE
Paste to upload

### ISSUE THREAD

**Issue by nicbou** [nicbou/markdown-notes#1]
It would be nice to be able to paste images (especially screenshots) rather than just upload them. There are no other markdown editors that support that at the moment, and it would be quite useful.

**Comment by vincentzhang** [comment]
One can start with the code snippet [here](http://stackoverflow.com/a/6338207)
It follows [Clipboard API spec](https://w3c.github.io/clipboard-apis/).
Busy with school now but I could find some time late May to get this implemented.

**Comment by AuHau** [comment]
So this feature was implemented in #46. As described in PR, the crossbrowser support for this is pretty horrible, but it should work for Chrome, Firefox and IE11. 

There is one rather big problem in the PR and it is that I am not using feature detection for distinguishing which technique to use, but I decide base on the name of browser (Chrome, Firefox...). I tried to figure out way how to write feature detection for it, but without success and also I was not able to find any hints on Internet.

This might introduce some problems with feature releases of browsers. Chrome should be alright, but others browsers might bring some trouble in future as they might introduce proper `paste` event support and deprecate the current technique.
